pandas/core/strings.py

```diff
@@ -150,14 +150,20 @@ def _map_stringarray(
         an ndarray.
 
     """
-    from pandas.arrays import IntegerArray, StringArray
+    from pandas.arrays import IntegerArray, StringArray, BooleanArray
 
     mask = isna(arr)
 
     assert isinstance(arr, StringArray)
     arr = np.asarray(arr)
 
-    if is_integer_dtype(dtype):
+    if is_integer_dtype(dtype) or is_bool_dtype(dtype):
+        constructor: Union[Type[IntegerArray], Type[BooleanArray]]
```

Comment:
```suggestion
        constructor: Type[Union[IntegerArray, BooleanArray]]
```
Optional but less verbose if you put the Union instead of the Type